TransformationPlayground/playground/transform_expert/expert.py

```diff
@@ -0,0 +1,139 @@
+import asyncio
+from dataclasses import dataclass
+import logging
+from typing import Any, Callable, Dict, List
+
+from botocore.config import Config
+from langchain_aws import ChatBedrockConverse
+from langchain_core.language_models import LanguageModelInput
+from langchain_core.messages import BaseMessage, SystemMessage, ToolMessage
+from langchain_core.runnables import Runnable
+
+from transform_expert.parameters import SourceVersion, TargetVersion, TransformType, TransformLanguage
+from transform_expert.prompting import get_system_prompt_factory
+from transform_expert.tools import ToolBundle, get_tool_bundle
+from transform_expert.utils.transforms import TransformTask
+
+
+logger = logging.getLogger("transform_expert")
+
+
+@dataclass
+class Expert:
+    llm: Runnable[LanguageModelInput, BaseMessage]
+    system_prompt_factory: Callable[[Dict[str, Any]], SystemMessage]
+    tools: ToolBundle
+
+def get_expert(source_version: SourceVersion, target_version: TargetVersion, transform_type: TransformType, transform_language: TransformLanguage) -> Expert:
+    logger.info(f"Building expert for: {source_version}, {target_version}, {transform_type}, {transform_language}")
+
+    # Get the tool bundle for the given transform language
+    tool_bundle = get_tool_bundle(transform_language)
+
+    # Define a boto Config to use w/ our LLM that's more resilient to long waits and frequent throttling
+    config = Config(
+        read_timeout=120, # Wait 2 minutes for a response from the LLM
+        retries={
+            'max_attempts': 20,  # Increase the number of retry attempts
+            'mode': 'adaptive'   # Use adaptive retry strategy for better throttling handling
+        }
+    )
+
+    # Define our Bedrock LLM and attach the tools to it
+    llm = ChatBedrockConverse(
+        model="anthropic.claude-3-5-sonnet-20240620-v1:0", # This is the older version of the model, could be updated
+        temperature=0, # Suitable for straightforward, practical code generation
+        max_tokens=4096,
```

Comment:
In the future, I expect this to be fully configurable by the user - what if they want to use their own LLM, for example, rather than bedrock?  We'll need to figure out what the right level of configurability is for initial user feedback.  On the token front - that is the maximum number of output tokens, not input.  4k tokens is a LOT of code, I would be highly surprised if we ever had a transform that needed anywhere close to that.  For reference, the ~450 lines of text in the OS 2.7 "knowledge" file is only ~3500 tokens.